src/Kirel.Identity.Client.Jwt/Providers/KirelJwtTokenAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using Kirel.Identity.Client.Interfaces;
+using Microsoft.AspNetCore.Components.Authorization;
+
+namespace Kirel.Identity.Client.Jwt.Providers;
+
+public class KirelJwtTokenAuthenticationStateProvider : AuthenticationStateProvider
+{
+    private readonly AuthenticationState _anonymousUserAuthenticationState;
+    private readonly IClientTokenService _tokenService;
+
+    public KirelJwtTokenAuthenticationStateProvider(IClientTokenService tokenService)
+    {
+        _tokenService = tokenService;
+        _anonymousUserAuthenticationState = new AuthenticationState(new ClaimsPrincipal(new ClaimsIdentity()));
+    }
+
+    /// <inheritdoc />
+    public override async Task<AuthenticationState> GetAuthenticationStateAsync()
+    {
+        var state = _anonymousUserAuthenticationState;
+        var token = await _tokenService.GetAccessTokenAsync();
+        if (!string.IsNullOrEmpty(token))
+            state = new AuthenticationState(new ClaimsPrincipal(new ClaimsIdentity(GetClaimsFromJwt(token), "JwtToken")));
+        
+        NotifyAuthenticationStateChanged(Task.FromResult(state));
```

Comment:
The call to the notify function is hidden in the get function, which can be confusing in the future. It seems to me that it is better to put the notification in a separate function

Reply:
Yes, that's a good idea.